tests/test_workflows.py

```diff
@@ -119,4 +119,34 @@ async def event_workflow(c: Context):
         assert s.w_output == "Hello from outside! B A"
 
 
-# TODO: test concurrent
+@async_test
+async def test_utils_methods(t: WorkflowTest):
+    @workflow()
+    async def utils_workflow(c: Context):
+        current_time = await c.utils.now()
+        rand_value = await c.utils.random(1, 100)
+        unique_id = await c.utils.uuid4()
+        return str(current_time), rand_value, str(unique_id)
+
+    backends = [
+        FSBackend("./logs"),
+        RedisBackend(host="localhost", port=6380, password="password"),
+    ]
+
+    for backend in backends:
+        start_time = datetime.now()
+        s = await t.step(backend).exec_workflow(utils_workflow)
+        end_time = datetime.now()
+
+        assert isinstance(s.w_output, list)
+        current_time, rand_value, unique_id = s.w_output
+
+        assert isinstance(current_time, str)
+        now_output = datetime.fromisoformat(current_time)
+        assert start_time <= now_output <= end_time
+
+        assert isinstance(rand_value, int)
+        assert 1 <= rand_value <= 100
```

Comment:
nitpick: This line only tests python's randint's behavior, not really the durable part of substantial.

But ultimately, all of these relies on `c.save` being correct so any scheme that can test any arbitrary save will do as a basis.